### PR TITLE
Feat/add logger interface

### LINKED ISSUE #7: Add support for a new generic logger type
Classes `AwsUploadModule` and `S3Uploader` both have required input args for a logger of type `winston.Logger`

To be more dynamic and allow for simpler logging (eg console log) if needed.
It is suggested that we replace the required type of `winston.Logger` with a logger interface

eg
```javascript
export interface ILogger {
  verbose: (message: string) => void;
  info: (message: string) => void;
  error: (message: string) => void;
}
``` 

Then one could implement such a logger when passing it to the module eg.

```javascript
class MyWinstonLogger implements ILogger {
  private logger: Logger;

  constructor(nodeEnv: string) {
    let level = "debug";
    if (nodeEnv && nodeEnv === "production") {
      level = "info"
    }
    this.logger = createLogger({
      level: level,
      transports: [ new transports.Console() ],
      format: combine(
        timestamp(),
        printf(({ level, message, timestamp }) => `${timestamp} ${level}: ${message}`)    
      ),
    });
  }

  info(message: string) {
    this.logger.info(message);
  }

  verbose(message: string) {
    this.logger.verbose(message);
  }

  error(message: string) {
    this.logger.error(message);
  }
}

class MySimpleLogger implements ILogger {
  info(message: string) {
   console.log(message);
  }

  verbose(message: string) {
   console.log(message);
  }

  error(message: string) {
    console.error(message);
  }
}

const awsUploadModule_1 = new AwsUploadModule(<args>,  new MyWinstonLogger()); 
const awsUploadModule_2 = new AwsUploadModule(<args>,  new MySimpleLogger()); 

```

### PR DESCRIPTION
**Pull request by Nfrederiksen**
This PR resolves #7 

Custom loggers which implement the `Logger` interface are to be passed through the AwsUploaderModule and S3Uploarder instead

### CODE REVIEW

File: lib/awsModule.ts [hunk 1]
```diff
@@ -26,15 +25,12 @@ export class AwsUploadModule implements IafUploadModule {
   onFileAdd = (filePath: string, readStream: Readable, contentType: string) => {
     this.fileName  = path.basename(filePath);
     try {
-      this.uploader.upload(readStream, this.fileName, process.env.AWS_FOLDER, contentType).then((res) => {
+      this.uploader.upload(readStream, this.fileName, process.env.AWS_FOLDER, process.env.AWS_FILE_CONTENT_TYPE).then((res) => {
         this.fileUploadedDelegate(res);
       });
     }
```
Comment (oscnord) [inline lib/awsModule.ts:31]: Should we not use `contentType` here?

Comment (Nfrederiksen) [reply lib/awsModule.ts:31]: i think i did a rebase error

File: package.json [hunk 2]
```diff
@@ -43,7 +43,9 @@
   "dependencies": {
     "@aws-sdk/client-s3": "^3.26.0",
     "@aws-sdk/lib-storage": "^3.27.0",
-    "@types/eyevinn-iaf": "^1.0.1",
-    "winston": "^3.3.3"
+    "@types/eyevinn-iaf": "^1.0.2"
+  },
+  "optionalDependencies": {
+    "fsevents": "^2.3.2"
```
Comment (oscnord) [inline package.json:49]: How is this handled when installing it (iaf-plugin-aws-s3) as a package?

Comment (Nfrederiksen) [reply package.json]: nvm, I deleted `node_modules` and `package-lock.json` and ran npm install and now the ci tests don't seem to need fsevents

**Comment by oscnord** [comment]
LGTM I'm not sure how this release should be handled because this will break the applications that currently uses this with a winston logger

**Review by oscnord — APPROVED** [review]

**Pull request merged**